app/serializers/api/product_serializer.rb

```diff
@@ -70,4 +38,12 @@ def variants
   def master
     options[:master_variants][object.id].andand.first
   end
+
+  def price
+    if options[:enterprise_fee_calculator]
+      object.master.price + options[:enterprise_fee_calculator].indexed_fees_for(object.master)
+    else
+      object.master.price_with_fees(options[:current_distributor], options[:current_order_cycle])
+    end
+  end
```

Comment:
I think this needs a spec in product_serializer_spec

Reply:
Ah yes. I just combined the old `ProductSerializer` and `CachedProductSerializer` here, but it looks like there were no specs for either of them.

Reply:
this is being set in the ProductsRenderer.products_json
you extracted it to a method with def enterprise_fee_calculator

Reply:
Serializer spec added :+1:

Reply:
did you remove it?
in case there's an enterprise fee in the OC, the product prices must include that fee.

it looks like variant.price_with_fees used EnterpriseFeeCalculator#fees_for while this uses EnterpriseFeeCalculator#indexed_fees_for 

I am not sure what's the difference, could be an optimized way of calculating the fees?

this commit https://github.com/openfoodfoundation/openfoodnetwork/commit/cf0031d9a560775596950bb1d655cc17b7191ae9
maybe it just added an optimized way of doing it, on the same class...
and we probably want to keep using it.

Reply:
I'll put it back in and add some test coverage in the Serializer spec.